### DISCUSSION TITLE
ignore missed time points on X axis for time serial/candlestick

### DISCUSSION THREAD

**Discussion by ZhengfengRao** [grafana/grafana#53706] [Plugins]
some time points may be missed in data source, such as weekends&vocations, it will display a gap on X axis for time serial or candlestick, it looks ugly. can we ignored the missed time points(date time)?

example as below:
![2022-08-13 14-57-47 的屏幕截图](https://user-images.githubusercontent.com/1232937/184473421-38e18186-d0c0-4089-99d0-7978b13e2b93.png)

**Comment by leeoniya** [comment]
not currently.

internally, the candlestick panel is built as an extension to the timeseries panel, which itself uses a continuous x axis. a discontinuous x axis would require some changes to how zooming works (you have to interpolate between adjacent points), how "nice" grid ticks are generated (data scanning), etc.

for a single dataset in a panel it's possible but not trivial.

things get quite tricky when you combine datasets that may potentially be discontinuous in different locations. e.g. the "weekend" in one timezone can be shifted by 24hrs in another. there are a lot of other hard questions that would have to be resolved in such cases. we could choose to say this mixed case will not be supported and simply fall back to a continuous x, too.

**Reply by ZhengfengRao** [reply]
thanks for your reply, i've got your point.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
